src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -961,25 +961,79 @@ def _match_host_names_from_cert(hostnames_from_cert, hostnames_in_webapp):
+
+    webapp_def = Site(location='')
     client = web_client_factory()
-    if is_valid_resource_id(plan):
-        plan = parse_resource_id(plan)['name']
-    location = _get_location_from_app_service_plan(client, resource_group_name, plan)
-    webapp_def = Site(server_farm_id=plan, location=location)
+    if consumption_plan_location is not None:
+        locations = list_consumption_locations()
+        location = next((l for l in locations if l['name'] == consumption_plan_location), None)
+        if location is None:
+            raise CLIError("Location is invalid. Use: az functionapp list-consumption-locations")
+        webapp_def.location = consumption_plan_location
+    else:
+        if is_valid_resource_id(plan):
+            plan = parse_resource_id(plan)['name']
+        location = _get_location_from_app_service_plan(client, resource_group_name, plan)
+        webapp_def.server_farm_id = plan
+        webapp_def.location = location
+
+    con_string = _validate_and_get_connection_string(resource_group_name, storage_account)
+
     webapp_def.kind = 'functionapp'
+    poller = client.web_apps.create_or_update(resource_group_name, name, webapp_def)
+    webapp = AppServiceLongRunningOperation()(poller)
+
+    # adding appsetting to site to make it a function
+    settings = ['AzureWebJobsStorage=' + con_string, 'AzureWebJobsDashboard=' + con_string,
+                'WEBSITE_NODE_DEFAULT_VERSION=6.5.0', 'FUNCTIONS_EXTENSION_VERSION=~1']
+
+    if consumption_plan_location is None:
+        update_site_configs(resource_group_name, name, always_on='true')
+    if consumption_plan_location is not None:
+        settings.append('WEBSITE_CONTENTAZUREFILECONNECTIONSTRING=' + con_string)
+        settings.append('WEBSITE_CONTENTSHARE=' + name.lower())
+
+    update_app_settings(resource_group_name, name, settings, None)
+
+    return webapp
+
+
+def _validate_and_get_connection_string(resource_group_name, storage_account):
     sa_resource_group = resource_group_name
     if is_valid_resource_id(storage_account):
         sa_resource_group = parse_resource_id(storage_account)['resource_group']
         storage_account = parse_resource_id(storage_account)['name']
-    poller = client.web_apps.create_or_update(resource_group_name, new_app_name, webapp_def)
-    webapp = AppServiceLongRunningOperation()(poller)
     storage_client = get_mgmt_service_client(StorageManagementClient)
-    keys = storage_client.storage_accounts.list_keys(sa_resource_group, storage_account).keys
+    storage_properties = storage_client.storage_accounts.get_properties(sa_resource_group,
+                                                                        storage_account)
+    error_message = ''
+    endpoints = storage_properties.primary_endpoints
+    sku = storage_properties.sku.name.value
+    allowed_storage_types = ['Standard_GRS', 'Standard_LRS', 'Standard_ZRS', 'Premium_LRS']
+
+    if (hasattr(endpoints, 'table') and not endpoints.table):
```

Comment:
Also, a question does function app require all 3 endpoints available or either one of them will do?